test/UnitTestFramework/UnitTestFramework.js

```diff
@@ -167,8 +191,120 @@ var testRunner = function testRunner() {
             }
 
             ++executedTestCount;
+        },
+
+        runAsyncTests: function runAsyncTests(testsToRun, options) {
+            ///<summary>Run provided tests that loads async code with WScript.LoadModule.<br/>
+            ///    source must be pre-processed so test results can be collected and tallied
+            ///    usage: WScript.LoadModule(testRunner.async(source, shouldFail))
+            ///
+            /// The 'testsToRun' is an object that has enumerable properties,<br/>
+            /// each property is an object that has 'name' and 'body' properties.
+            ///</summary>
+
+            asyncTest.count = new Array(testsToRun.length).fill(0);
+            asyncTest.error = new Array(testsToRun.length).fill("");
+
+            if (options) {
+                if (options.hasOwnProperty("verbose")) {
+                    _verbose = options.verbose;
+                }
+            }
+
+            for (var i in testsToRun) {
+                var isRunnable = typeof testsToRun[i] === objectType;
+                if (isRunnable) {
+                    this.runAsyncTest(i, testsToRun[i].name, testsToRun[i].body);
+                }
+            }
+
+            if (_verbose || executedTestCount != passedTestCount) {
+                helpers.writeln("Summary of async tests: total load attempts: ", executedTestCount,
+                  "; loaded: ", passedTestCount, "; failed: ", executedTestCount - passedTestCount);
+            }
+
+            Promise.all(asyncTest.promise.map(Promise.all, Promise)).then( function() {
+                passedTestCount = 0;
+                for (var i = 0; i < asyncTest.count.length; i++) {
+                    var passed = testRunner.asyncTestHasNoErr(i);
+                    if (_verbose) {
+                        helpers.writeln("*** Result of test #", i + 1, " (", i, "): ", testsToRun[i].name);
+                        if (!passed) helpers.writeln(asyncTest.error[i]);
+                        helpers.writeln(passed ? "PASSED" : "FAILED");
+                    }
+                    !passed || passedTestCount++;
+                }
+                if (_verbose || executedTestCount != passedTestCount) {
+                    helpers.writeln("Summary of tests: total executed: ", executedTestCount,

```

Comment:
There's no particular reason other than consistency with the older code.

---
In reply to: [110987334](https://github.com/Microsoft/ChakraCore/pull/2822#discussion_r110987334) [](ancestors = 110987334)